### DISCUSSION TITLE
Format `.html` files in VS Code?

### DISCUSSION THREAD

**Discussion by vincerubinetti** [Shopify/prettier-plugin-liquid#162] [Q&A]
I've followed [the instructions here](https://github.com/Shopify/prettier-plugin-liquid/wiki/Use-it-in-your-editor#from-the-prettier-extension) to use the regular Prettier extension with this plugin, but it seems like it only works on `.liquid` files. I'm writing a Jekyll website template that uses `.html` files. Is there any way to configure VS Code settings or the plugin to run on `.html` files?

FYI, changing the `.html` extension temporarily to `.liquid` works correctly and the plugin is able to format the file just fine. I tried the Shopify Liquid extension method, and temporarily set it as the default formatter for html:

```json
  "[html]": {
    "editor.defaultFormatter": "Shopify.theme-check-vscode"
  },
```

but trying to format gives an error "Extension 'Shopify Liquid' is configured as formatter but it cannot format 'HTML'-files.".

**Comment by charlespwd** [comment]
🤔 Have you tried the solution [here](https://stackoverflow.com/questions/29973619/how-to-associate-a-file-extension-with-a-certain-language-in-vs-code/51228725#51228725)?

I feel like changing the file association from HTML to Liquid would solve your problem.

**Comment by chrisrhymes** [comment]
I've added the following as workspace settings (in .vscode/settings.json) and it seems to now auto detect as liquid in Jekyll html files and markdown files in VS Code. 

```json
{
  "files.associations": {
    "*.html": "liquid",
    "*.md": "liquid"
  },
  "[liquid]": {
    "editor.defaultFormatter": "Shopify.theme-check-vscode",
    "editor.formatOnSave": true
  }
}
```
